### ISSUE TITLE
Feature request: misuse of 1.0 listeners syntax

### ISSUE THREAD

**Issue by web-padawan** [Polymer/polymer-linter#9]
It would be nice to have a check for using `id` of element which is located inside of the `dom-if` template in `listeners` object. As of 1.7.0, this causes errors like this: 
```
polymer.html:540 Uncaught TypeError: Invalid value used as weak map key
```

**Comment by rictic** [comment]
I see, so with an element whose template is like:

```html
<template>
  <template is="dom-if" if="whatever">
    <div id="foo"></div>
  </template>
</template>
```

And it's declared like:

```javascript
Polymer({
  is: 'example-elem',
  listeners: {
    'foo.tap': 'onFooTapped',
  },
  onFooTapped() { /* ... */ }
});
```

This doesn't work because the div doesn't exist when the element is first stamped.

**Comment by web-padawan** [comment]
Another example could be `this.$.elementId` calls check (linter should throw error if element does not have a child which such `id` in its template). Had a typo like this today 😢

**Comment by davidmaxwaterman** [comment]
...and I would imagine it might be an easy task to add a check for properties/methods on such - eg this.$.elementId.open() - check there's a method open() on the element.
